tiledb/sm/serialization/query.cc

```diff
@@ -1528,7 +1528,7 @@ Status query_from_capnp(
               query_buffer.original_validity_vector_size_;
         }
       }
-    } else if (query_type == QueryType::WRITE || type == QueryType::READ) {
+    } else if (query_type == QueryType::WRITE) {
```

Comment:
Do we even need the differentiation between the read and the writes?

Reply:
Maybe we don't anymore after this fix, but with @robertbindar we took a conscious decision to not refactor any further at them moment in lack of proper test coverage. We will tackle query serialization refactoring as a separate task, there's already https://app.shortcut.com/tiledb-inc/story/25026/refactor-query-deserialization-logic-specific-to-attr-copy-state-nullptr open for a part of it, and it will be probably used to clean up all this code that makes little sense anymore and adds up unecessary complexity.

Reply:
Ideally we don't need differentiation, we should have only serialization and deserialization logic threads, no rest_client and CAPI calls logic (via copy_state/attr_copy_state), no read and write separate logic, but it's just scary to remove anything from there, most probably the very reason why the code ended up in this shape, everybody just added.

We are now in a much better shape with the extended coverage from global order writes, open v2, etc via serialization wrappers, but still no major refactoring should be done unless either:
a) the rest server mimic project ends up in the codebase and all coverage becomes also serialization coverage.
b) there is special time dedicated to 1/2 engineers, no interuptions, so that they can load in memory all the spaghetti edge cases modeled in the query code and spit out a better version of the code that covers all of them (at least the ones that are still valid as I have a feeling lots of existing code in there model nothing, just fixes other hacks happening earlier in the call stack).

Reply:
Cool let's not do it here for now.